job-server/src/spark.jobserver/JobManagerActor.scala

```diff
@@ -153,7 +166,17 @@ class JobManagerActor(dao: JobDAO,
                        rddManagerActor: ActorRef): Option[Future[Any]] = {
     var future: Option[Future[Any]] = None
     breakable {
-      val lastUploadTime = dao.getLastUploadTime(appName)
+      import akka.pattern.ask
+      import akka.util.Timeout
+      import scala.concurrent.duration._
+      import scala.concurrent.Await
+
+      val daoAskTimeout = Timeout(3 seconds)
+      val resp = Await.result(
+        (daoActor ? JobDAOActor.GetLastUploadTime(appName))(daoAskTimeout).mapTo[JobDAOActor.LastUploadTime],
+        daoAskTimeout.duration)
+
+      val lastUploadTime = resp.lastUploadTime
```

Comment:
I don't recommend using Await in an actor. It looks like you can do a for comp here with the Future.